ublox/src/parser.rs

```diff
@@ -496,7 +525,109 @@ impl<T: UnderlyingBuffer> ParserIter<'_, T> {
                 self.buf.drain(2);
                 continue;
             }
-            return self.extract_packet(pack_len);
+            return extract_packet_ubx(&mut self.buf, pack_len);
+        }
+        None
+    }
+}
+
+/// Iterator over data stored in `Parser` buffer
+pub struct UbxRtcmParserIter<'a, T: UnderlyingBuffer> {
+    buf: DualBuffer<'a, T>,
+}
+
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub struct RtcmPacketRef<'a> {
+    pub data: &'a [u8],
+}
+
+fn extract_packet_rtcm<'a, 'b, T: UnderlyingBuffer>(
+    buf: &'b mut DualBuffer<'a, T>,
+    pack_len: usize,
+) -> Option<Result<AnyPacketRef<'b>, ParserError>> {
+    if !buf.can_drain_and_take(0, pack_len + 3) {
+        if buf.potential_lost_bytes() > 0 {
+            // We ran out of space, drop this packet and move on
+            // TODO: shouldn't we drain pack_len + 3?
+            buf.drain(2);
+            return Some(Err(ParserError::OutOfMemory {
+                required_size: pack_len + 2,
+            }));
+        }
+        return None;
+    }
+
+    let maybe_data = buf.take(pack_len + 3);
+    match maybe_data {
+        Ok(data) => Some(Ok(AnyPacketRef::Rtcm(RtcmPacketRef::<'b> { data }))),
+        Err(e) => Some(Err(e)),
+    }
+}
+
+impl<T: UnderlyingBuffer> UbxRtcmParserIter<'_, T> {
+    fn find_sync(&self) -> NextSync {
+        for i in 0..self.buf.len() {
+            if self.buf[i] == SYNC_CHAR_1 {
+                return NextSync::Ubx(i);
+            }
+            if self.buf[i] == RTCM_SYNC_CHAR {
+                return NextSync::Rtcm(i);
+            }
+        }
+        NextSync::None
+    }
+
+    #[allow(clippy::should_implement_trait)]
+    /// Analog of `core::iter::Iterator::next`, should be switched to
+    /// trait implementation after merge of https://github.com/rust-lang/rust/issues/44265
```

Comment:
The linked issue has been resolved btw 😄

Reply:
Hahaha. Then we should switch it :grin: